cli/util/upload_bin.go

```diff
@@ -467,41 +453,19 @@ func searchObjects(ctx context.Context, p poolWrapper, containerID cid.ID, accou
 }
 
 // uploadObj uploads object to the container using provided settings.
-func uploadObj(ctx context.Context, p poolWrapper, signer user.Signer, owner util.Uint160, containerID cid.ID, objData []byte, attrs []object.Attribute, homomorphicHashingDisabled bool) (oid.ID, error) {
+func uploadObj(ctx context.Context, p poolWrapper, signer user.Signer, owner util.Uint160, containerID cid.ID, objData []byte, attrs []object.Attribute) (oid.ID, error) {
 	var (
 		ownerID          user.ID
 		hdr              object.Object
-		chSHA256         checksum.Checksum
-		chHomomorphic    checksum.Checksum
-		v                = new(version.Version)
 		prmObjectPutInit client.PrmObjectPutInit
 		resOID           = oid.ID{}
 	)
 
 	ownerID.SetScriptHash(owner)
 	hdr.SetPayload(objData)
-	hdr.SetPayloadSize(uint64(len(objData)))
 	hdr.SetContainerID(containerID)
 	hdr.SetOwnerID(&ownerID)
 	hdr.SetAttributes(attrs...)
-	hdr.SetCreationEpoch(1)
-	v.SetMajor(1)
-	hdr.SetVersion(v)
-	if !homomorphicHashingDisabled {
-		checksum.Calculate(&chHomomorphic, checksum.TZ, objData)
-		hdr.SetPayloadHomomorphicHash(chHomomorphic)
-	}
-	checksum.Calculate(&chSHA256, checksum.SHA256, objData)
-	hdr.SetPayloadChecksum(chSHA256)
-
-	err := hdr.SetIDWithSignature(signer)
-	if err != nil {
-		return resOID, err
-	}
-	err = hdr.CheckHeaderVerificationFields()
-	if err != nil {
-		return resOID, err
-	}
```

Comment:
You've removed these setters, but I don't see that anything is changed on the object put code. Who's setting and verifying these fields now?

Reply:
Basicly i was looking at example https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/pool@v1.0.0-rc.12#Pool.ObjectPutInit and now we are not setting those fields so we should not check them . 

Checking where this fields are checked.

Reply:
> You've removed these setters, but I don't see that anything is changed on the object put code. Who's setting and verifying these fields now?

storage node does in this case. The correctness guaranteed in this case, any checks are redundant. Moreover, this check will definitely fail here

Reply:
> Moreover, this check will definitely fail here

We used this code before, it worked OK. But anyway, it'll be removed now.